source/common/listener_manager/fcds_api.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "envoy/config/core/v3/config_source.pb.h"
+#include "envoy/config/listener/v3/listener_components.pb.h"
+#include "envoy/config/listener/v3/listener_components.pb.validate.h"
+#include "envoy/config/subscription.h"
+#include "envoy/server/listener_manager.h"
+
+#include "source/common/config/subscription_base.h"
+#include "source/common/init/target_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * FCDS API implementation that fetches via Subscription.
+ */
+class FcdsApiImpl : public FcdsApi,
+                    Envoy::Config::SubscriptionBase<envoy::config::listener::v3::FilterChain>,
```

Comment:
Is an FcdsApiImpl object a type Subscription?
This seems a bit strange and an is-a relationship probably not the right thing here.